packages/calypso-e2e/src/lib/pages/plans-page.ts

```diff
@@ -141,7 +147,10 @@ export class PlansPage {
 		// and https://github.com/Automattic/wp-calypso/pull/64421#discussion_r892589761.
 		await Promise.all( [
 			this.page.waitForLoadState( 'networkidle' ),
-			this.page.waitForResponse( /.*active-promotions.*/ ),
+			// don't wait for the response if it's already done in the past
+			this.receivedActivePromotions
+				? Promise.resolve()
+				: this.page.waitForResponse( /.*active-promotions.*/ ),
```

Comment:
**suggestionI**: think we can remove the wait for `active-promotion` instead. 

I had initially selected this request because I thought it was universal between Simple/Atomic for all users but looks like it isn't the case. Rather than introduce a new class attribute to set and conditional behavior in tests (which I discourage), it is better to find a request that is universal _and_ fires towards the end of the PlansPage loading.

What do you think?

Reply:
Yes I totally agree. What we if we don't wait for any requests at all?

Reply:
> What if we don't wait for any requests at all?

That's fine too. The reason why there's a wait here in the first place is that the "legacy" Plans page has navtabs that end up swallowing up the clicks if things are still loading. Annoyingly, waiting for `networkidle` to fire does not guarantee that everything is actually loaded. 

As far as I can tell, the only script using `PlansPage.clickTab` now is the [`tos-screenshot`](https://github.com/Automattic/wp-calypso/blob/b0f72481b942cd0da813ae8dd3111c0a51a73ea3/test/e2e/specs/martech/tos-screenshots__checkout.ts) file now.

Reply:
I realized my previous message didn't really offer any concrete response to your question. 

Personally, I'd say let's remove the wait on the network call and just let `networkidle` event handle the waiting. If the ToS script starts failing, then we can think about a solution then.